### DISCUSSION TITLE
logrotation for nats-server logs

### DISCUSSION THREAD

**Discussion by zakk616** [nats-io/nats-server#5800] [Q&A]
I have tried to configure log rotation for nats-server on one of my cluster nodes using `logrotate` , below is my nats logrotate config:

```
/var/log/nats/nats-server-*.log.* {
    daily
    missingok
    rotate 1
    compress
    notifempty
    create 640 root root
    sharedscripts
    postrotate
        /bin/systemctl reload nats-server > /dev/null 2>/dev/null || true
    endscript
    maxage 1
}
```
and tested it in dry and force mode with below commands:
```sudo logrotate -d /etc/logrotate.d/nats```
```sudo logrotate -f /etc/logrotate.d/nats```

what it does is, compresses all existing logs and keep the old logs with size zero and compressed files.
![image](https://github.com/user-attachments/assets/3573d03b-a737-4d69-b46a-cfe70d189c57)

what I need is to delete the older than on day files and compress today's logs. 

any idea ?

**Comment by ripienaar** [comment]
Use `copytruncate` and not `create ....`
